### DISCUSSION TITLE
Anything about ur suggestion can be posted here!

### DISCUSSION THREAD

**Discussion by K2Lin-Daniel** [K2Lin-Daniel/K2Lin_Tool#2] [Ideas]
![](https://img.shields.io/github/license/K2Lin-Daniel/K2Lin_Tool?label=License&style=plastic)

**Comment by K2Lin-Daniel** [comment]
If u think Script recoil is tooo Weak or Strong for ya,, u can customize it in config and use G10(Shift the wheel to the right) to cycle between 6 config slot:)

**Reply by K2Lin-Daniel** [reply]
well , i'll take a look about this :)

**Reply by K2Lin-Daniel** [reply]
OK I've figured out ..It works but maybe needs to keep the original bindings in GHUB like my default settings for Automatic active button is G5 and G5 button binds "Forward" in default. Keep it and it will work/
Another news: I will release a new version that can modify Activation of Rapid Fire binding button

**Reply by K2Lin-Daniel** [reply]
We've figured out a small bug in our code :(
Now it's fixed! try out the newest V10 version!

**Comment by KLZ-BRAZZA** [comment]
Hello, the code is very good .. I noticed that it sometimes works, sometimes it doesn't .. I have a wireless g502 ... only problem is having to keep opening the program to leave it in the correct indentation ... if the code was apparent on the screen would help a lot ! thank you again, you are great

**Reply by K2Lin-Daniel** [reply]
Sorry for replying so late! Are you using a Applications Config? Try to set it to Persistent one. it may helps:)

**Comment by KLZ-BRAZZA** [comment]
I currently use this code and it is very good ...

local recoil_mode  -- possible values: false, 8, 7, 6
 local recoil_loops = {
       [8] = {
          {x=-1, y=2, delay=7},
          {x=1, y=2, delay=7},
          {x=-1, y=3, delay=8},
          {x=1, y=3, delay=5},
       },
       [7] = {
          {x=0, y=5, delay=5},
          {x=0, y=6, delay=7},
          {x=0, y=7, delay=6},
          {x=0, y=8, delay=4},
          {x=0, y=9, delay=6},
          {x=0, y=11, delay=5},
       },
       [6] = {
          {x=0, y=8, delay=50},
          {x=0, y=8, delay=50},
          {x=0, y=8, delay=50},
          {x=0, y=8, delay=50},
          {x=0, y=9, delay=50},
          {x=0, y=11, delay=50},
       },
    }
    
    function OnEvent(event, arg)
       if event == "PROFILE_ACTIVATED" then
          OutputLogMessage("PROFILE_ACTIVATED")
          EnablePrimaryMouseButtonEvents(true)
       elseif event == "PROFILE_DEACTIVATED" then
          OutputLogMessage("PROFILE_DEACTIVATED")
          ReleaseMouseButton(1) -- to prevent it from being stuck on
       elseif event == "MOUSE_BUTTON_PRESSED" and recoil_loops[arg] then
          recoil_mode = recoil_mode ~= arg and arg
          OutputLogMessage("Recoil mode is now "..tostring(recoil_mode))
       elseif event == "MOUSE_BUTTON_PRESSED" and arg == 1 and recoil_mode then
          local loop = recoil_loops[recoil_mode]
          local j = 0
          repeat
             j = j % #loop + 1
             if IsMouseButtonPressed(3) then  -- only when RMB pressed
                MoveMouseRelative(loop[j].x, loop[j].y)
             end
             Sleep(loop[j].delay)
          until not IsMouseButtonPressed(1)
       end
     end
